src/mono/mono/mini/aot-compiler.c

```diff
@@ -5252,8 +5255,14 @@ MONO_RESTORE_WARNING
 				mono_error_assert_ok (error);
 
 				add_method (acfg, wrapper);
-				if (export_name)
+				if (export_name) {
 					g_hash_table_insert (acfg->export_names, wrapper, export_name);
+#ifdef TARGET_APPLE_MOBILE
```

Comment:
Is this preferred, should we pass in the symbol format from the MonoAOTCompiler task, or some other way?

Reply:
There is a acfg->user_symbol_prefix field which is already set correctly based on the platform.